datafusion/core/src/datasource/physical_plan/parquet.rs

```diff
@@ -224,6 +224,64 @@ mod tests {
         )
     }
 
+    #[tokio::test]
+    async fn test_pushdown_with_missing_column_in_file() {
+        let c1: ArrayRef = Arc::new(Int32Array::from(vec![1, 2, 3]));
+
+        let file_schema =
+            Arc::new(Schema::new(vec![Field::new("c1", DataType::Int32, true)]));
+
+        let table_schema = Arc::new(Schema::new(vec![
+            Field::new("c1", DataType::Int32, true),
+            Field::new("c2", DataType::Int32, true),
```

Comment:
Can you please also add tests (or ensure they already exist) for:
1. When the type of c1 is different than c2 (so like `Utf8` for example)
2. When the missing column is in the middle of the schema (so like a table schema with `c1, c2, c3` and a file schema like `c1, c3` or something)
3. When the file schema has columns out of order (so like a table schema with `c1, c2, c3` and a file schema like `c3, c3`)